onnxruntime/core/providers/cuda/reduction/reduction_functions.cu

```diff
@@ -286,23 +285,15 @@ bool is_matrix_row_reduction(
   if (cudnn_reduce_op != CUDNN_REDUCE_TENSOR_ADD)
     return false;
 
-  // Check if all but the last axis are reduced. For example, reducing
```

Comment:
Why are those checks replaced by `axes.sizes() < 1`? Are they equivalent?

Reply:
an extended check is performed in the PrepareForReduce as it needs to be done anyway for the stride calculation, which is needed before this function call. The prepare_reduce_metadata.contiguous_axes stores this result.